Fix #10069 - Fix issue with workflow/report conditions not saving values when moving between modules

JackBuchanan opened the pull request:
Add functionality to loop through relationship path to pass in correct module to fixUpFormatting

<!--- Provide a general summary of your changes in the Title above -->
<!--- Please be aware that as of the 31st January 2022 we no longer support 7.10.x.
New PRs to hotfix-7.10.x will be invalid. If your fix is still applicable to 7.12.x, 
please create the pull request to the hotfix branch accordingly. -->


## Description
Add in new loop that goes through the relationship path (e.g. Accounts->Contacts->Calls) so that the correct module is passed into fixUpFormatting.

Previously the report/workflow base module is always passed into this function and if the relationship has a custom field that does not exist (e.g. Accounts->Contacts->Calls:custom_date) 

Based off of fix: https://github.com/salesagility/SuiteCRM/pull/10070 and expanded to cover reports and additional relationships

Credit to user https://github.com/re8260 for the initial fix.

<!--- Describe your changes in detail -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here unless your commit contains the issue number -->
<!--- Ensure that all code ``` is surrounded ``` by triple back quotes. This can also be done over multiple lines -->

## Motivation and Context
Fixes issues with date fields from related modules not saving
<!--- Why is this change required? What problem does it solve? -->

## How To Test This
Use steps from issue https://github.com/salesagility/SuiteCRM/issues/10069
<!--- Please describe in detail how to test your changes. -->

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist
<!--- Go over all the following points and check all the boxes that apply. --->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! --->
- [x] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [ ] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

<!--- Your pull request will be tested via Travis CI to automatically indicate that your changes do not prevent compilation. --->

<!--- If it reports back that there are problems, you can log into the Travis system and check the log report for your pull request to see what the problem was. --->

JackBuchanan committed:
Fix #10069 - Fix issue with workflow/report conditions not saving values when moving between modules

Add functionality to loop through relationship path to pass in correct module to fixUpFormatting

serhiisamko091184 commented:
Hello @JackBuchanan! 

Thanks for your PR! 

Regards,
Serhii

johnM2401 reviewed (dismissed):
LGTM
Tried Date and Datetime fields on various relationships, including one 3-deep.
Tried on WF and Reports, with each Date and Time format.
![image](https://github.com/user-attachments/assets/018a0c52-6a8a-435f-9d60-b475147b1118)

JackBuchanan committed:
Fix #10069 - Fix issue with workflow/report conditions not saving values when moving between modules

Add functionality to loop through relationship path to pass in correct module to fixUpFormatting
